specification/refarch.adoc

```diff
@@ -3,142 +3,184 @@
+The TSM enforces isolation between TVMs by using the G-stage page table. Figure
+1 below shows the abstract model of isolation:
+
+* Hart operating with the confidential supervisor domain context has MTT
+permissions to access Confidential and Non-Confidential memory
+* Hart not operating in a Confidential supervisor domain has access permissions
+only for Non-confidential Memory
 
 [caption="Figure {counter:image}: ", reftext="Figure {image}"]
 [title= "Confidential Memory conversion"]
 image::img_7.png[]
 
-Memory with Confidential PMA may be associated with a unique memory encryption 
-key (and similar IO/fabric protection policies). 
-Non-confidential memory is assigned by the VMM - the TSM and TSM-driver are 
-expected to manage the Confidentiality PMA by programming a Memory Tracking Table (MTT). 
-The desired security properties of memory tracking are discussed below - implementations 
-are free to choose the format and structure of the memory tracking table/structures. 
-The TSM manages finer-granular (page-based) allocation from Confidential memory regions 
-(enforced by the memory tracking hardware) using the G stage page tables.
-
-Four aspects of memory isolation are impacted due to this dynamic configurable confidential PMA: 
+[NOTE]
+====
+To also provide physical memory protection, memory accessible to the
+confidential supervisor domain via MTT may also be associated with a unique
+memory encryption key. Additionally each TVM may also be associated with a
+unique memory encryption key. These additional protection aspects are platform
+and implementation dependent.
+====
+
+Confidential and non-confidential memory are both always assigned by the VMM -
+the TSM and TSM-driver are expected to manage the isolation for confidential
+memory by programming the Memory Tracking Table (MTT). The desired security
+properties of memory tracking are discussed below. The TSM manages
+finer-granular (page-based) allocation from Confidential memory regions
+(enforced by the memory tracking hardware) using the G-stage page table.
+
+Four aspects of memory isolation are impacted due to this dynamic configurable
+property of the MTT:
 
 ==== Address Translation/Page Walk
-The figure 4 below describes a reference model for memory tracking lookup where 
-the physical address derived from the first and guest stage nested page walk is 
-looked up to derive a confidential | non-confidential physical memory attribute. 
-This lookup should be performed for the page sizes per the paging modes supported by the hart.
+The figure 2 below describes a reference model for memory tracking lookup where
+the physical address derived from the first and guest stage nested page walk is
+looked up via the MTT configured for the active supervisor domain to get the
+access permissions for the physical address. This lookup should be performed
+for the page sizes per the paging modes supported by the hart.
```

Comment:
```suggestion
access permissions for the physical address. This lookup should be performed 
for each implicit and explicit memory access and per the paging sizes/modes 
supported by the hart.
```

Reply:
"and per paging sizes/modes..." can be removed.